src/components/Wizard/Patterns/WizardPattern.js

```diff
@@ -0,0 +1,196 @@
+
+  return (
+    <Modal
+      show={show}
+      onHide={onHide}
+      onExited={onExited}
+      dialogClassName="modal-lg wizard-pf"
+    >
+      <Wizard>
+        <Modal.Header>
+          <button
+            className="close"
+            onClick={onHide}
+            aria-hidden="true"
+            aria-label="Close"
+          >
+            <Icon type="pf" name="close" />
+          </button>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body className="wizard-pf-body clearfix">
+          <WizardPatternBody
+            loadingTitle={loadingTitle}
+            loadingMessage={loadingMessage}
+            loading={loading}
+            steps={steps}
+            activeStepIndex={activeStepIndex}
+            activeStepStr={activeStepStr}
+            goToStep={goToStep}
+            nextStepDisabled={nextStepDisabled}
+            stepButtonsDisabled={stepButtonsDisabled}
+          />
+        </Modal.Body>
+        <Modal.Footer className="wizard-pf-footer">
+          <Button bsStyle="default" className="btn-cancel" onClick={onHide}>
+            {cancelText}
+          </Button>
+          <Button
+            bsStyle="default"
+            onClick={onBackClick}
+            disabled={prevStepUnreachable}
+          >
+            <Icon type="fa" name="angle-left" />
+            {backText}
+          </Button>
+          <Button
+            bsStyle="primary"
+            onClick={onFinalStep ? onHide : onNextClick}
+            disabled={nextStepUnreachable}
+          >
+            {onFinalStep ? (
+              closeText
+            ) : (
+              <React.Fragment>
+                {nextText}
+                <Icon type="fa" name="angle-right" />
+              </React.Fragment>
+            )}
+          </Button>
+        </Modal.Footer>
+      </Wizard>
+    </Modal>
+  );
+};
+
+WizardPattern.propTypes = {
+  activeStepIndex: PropTypes.number.isRequired,
+  show: PropTypes.bool,
+  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  onHide: PropTypes.func,
+  onExited: PropTypes.func,
+  onBack: PropTypes.func,
+  onNext: PropTypes.func,
+  onStepChanged: PropTypes.func,
+  loadingTitle: PropTypes.string,
+  loadingMessage: PropTypes.string,
+  loading: PropTypes.bool,
+  cancelText: PropTypes.string,
+  backText: PropTypes.string,
+  nextText: PropTypes.string,
+  closeText: PropTypes.string,
+  steps: PropTypes.arrayOf(PropTypes.shape(wizardStepShape)),
+  nextStepDisabled: PropTypes.bool,
+  stepButtonsDisabled: PropTypes.bool
+};
+
+WizardPattern.defaultProps = {
+  show: false,
+  title: '',
+  onHide: noop,
+  onExited: noop,
+  onBack: noop,
+  onNext: noop,
+  onStepChanged: noop,
+  loadingTitle: 'Loading Wizard...',
+  loadingMessage: 'Loading...',
+  loading: false,
+  cancelText: 'Cancel',
+  backText: 'Back',
+  nextText: 'Next',
+  closeText: 'Close',
+  steps: [{ title: 'General', render: () => <p>General</p> }],
```

Comment:
`steps` seems to be the most critical bit of data that defines the wizard component - I'd suggest to make the prop required.

This is similar to e.g. `Button` component not providing any fallback for its content (children).

Reply:
i was thinking about this... what if you have some strange dynamic wizard where steps are unknown upfront... unlikely but it _could_ happen...so for now i think it OK to just return `null` ...

Reply:
i made a few tweaks to this in my commits..

Reply:
@priley86 is right, in this case I imagine that `steps` may be undefined while `loading` is true, in some cases. Maybe in that case you should still be forced to pass an empty array of steps?

Reply:
`null` or empty array could imply loading state, since we don't have a notion of "loaded but empty" kind of state.